flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/resources/FlinkPod.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.kubeclient.resources;
+
+import org.apache.flink.configuration.Configuration;
+
+import io.fabric8.kubernetes.api.model.ContainerStatus;
+import io.fabric8.kubernetes.api.model.Pod;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represent FlinkPod resource in kubernetes.
+ */
+public class FlinkPod extends Resource<Pod> {
```

Comment:
I feel name `KubernetesPod` would be more descriptive than `FlinkPod`, since this code is in Flink project and it actually represents kubenetes components. 
The same applies to other components, including FlinkConfigMap, FlinkDeployment and FlinkService.

Reply:
After reconsidering and discussion with other guys, `KubernetesPod` is better. I will rename the resources.